docs/advanced/custom-middlewares.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Custom middlewares
+---
+
+The aim of this document is to show you how you can add custom middlewares to a Dara app.
+
+## Defining a middleware
+
+A middleware can be a custom class or a function. A simple middleware class could be defined as the following
```

Comment:
I think the only thing missing is why would someone want to do that? Can be a short sentence or two just listing out potential reasons or what does it allow you to accomplish. Perhaps also link to FastAPI middleware docs for more details